tools/go-cli/Dockerfile

```diff
@@ -34,33 +34,70 @@ RUN mkdir /src/github.com/go-whisk-cli/build
 
 # The BUILDTIME argument is used as the CLI build time property (wsk property get --all)
 # Disabled for now to speed build times
-#ARG BUILDTIME
+ARG LOCAL_BUILD
+ARG LOCAL_OS
+ARG LOCAL_ARCH
+
+RUN echo $LOCAL_ARCH
 
 # Build the Go wsk CLI binaries
-ENV GOOS=windows
 ENV EXEC=wsk.exe
-ENV ARCHS="386 amd64"
-RUN for GOARCH in $ARCHS; do \
+RUN if [ "$LOCAL_BUILD" = "true" ]; then \
+      if [ "$LOCAL_OS" = "windows" ] ; then \
+        export GOOS=$LOCAL_OS && \
+        export ARCHS=$LOCAL_ARCH; \
+      else \
+        unset GOOS && \
+        unset ARCHS; \
+      fi; \
+    else \
+      export GOOS=windows && \
+      export ARCHS="386 amd64"; \
+    fi && \
```

Comment:
I think that's the idea, the PR is `wip` so I give a chance to @dubeejw to have it ready for review.
here is POC: on how would look like running from a script, and the script can even be smaller
https://github.com/openwhisk/openwhisk/compare/master...csantanapr:gocli_build_script